server/handler.go

```diff
@@ -518,11 +523,13 @@ func (h *Handler) AddTransferRegionOperator(regionID uint64, storeIDs map[uint64
 	}
 
 	peers := make(map[uint64]*metapb.Peer)
-	for id := range storeIDs {
-		peers[id] = &metapb.Peer{StoreId: id}
+	roles := make(map[uint64]placement.PeerRoleType)
+	for id, peerRole := range storeIDs {
+		peers[id] = &metapb.Peer{StoreId: id, Role: peerRole.MetaPeerRole()}
+		roles[id] = peerRole
 	}
 
-	op, err := operator.CreateMoveRegionOperator("admin-move-region", c, region, operator.OpAdmin, peers)
+	op, err := operator.CreateMoveRegionOperator("admin-move-region", c, region, operator.OpAdmin, peers, roles)
```

Comment:
I think we can pass only `roles` and create `metapb.Peer` inside the function.

Reply:
Done.